sos/cleaner/parsers/username_parser.py

```diff
@@ -26,8 +26,8 @@ class SoSUsernameParser(SoSCleanerParser):
     map_file_key = 'username_map'
     regex_patterns = []
 
-    def __init__(self, config, skip_cleaning_files=[]):
-        self.mapping = SoSUsernameMap()
+    def __init__(self, config, skip_cleaning_files=[], workdir='/tmp'):
```

Comment:
The `workdir=`/tmp` is here due to https://github.com/sosreport/sos/blob/main/tests/unittests/cleaner_tests.py#L33-L38 where we need to pass some value.

Calls from the cleaner itself always use `os.path.dirname(self.opts.map_file)` so we just need some safe fallback for tests.

I am not sure if `/tmp` is a good dir for all distros, and if we should set it here or in the tests - let me know your thoughts.

Reply:
it's ok from the Debian/Ubuntu side, as we use `/tmp` already for all our configs as a patch (although I'll be changing that in the policy instead)

Reply:
When I run avocado tests in rhel, I get them placed in /var/tmp. I imagine it uses the _tmp_dir inside policy. 
Since @arif-ali has added now the same var in the debian policy via e4c50c2526f5f2a83a0302c5bb986ca14e00ecf9, maybe it would be better to use policy.get_tmp_dir() to get the right tmp directory depending on distro?

Reply:
We had the config option specific in the sos.conf it was being placed in `/tmp` already. I only just moved it in the polict instead, which made more sense.

I'd agree, we ought to use the policy configuration (if we can), then you know it will be consistent based on those parameters rather than a static entry on the code.